sdk/core/Azure.Core.Amqp/src/Shared/AmqpAnnotatedMessageConverter.cs

```diff
@@ -311,10 +311,20 @@ public static AmqpAnnotatedMessage FromAmqpMessage(AmqpMessage source)
             {
                 if (source.Properties.AbsoluteExpiryTime.HasValue)
                 {
-                    message.Properties.AbsoluteExpiryTime =
-                        source.Properties.AbsoluteExpiryTime >= DateTimeOffset.MaxValue.UtcDateTime
-                            ? DateTimeOffset.MaxValue
-                            : source.Properties.AbsoluteExpiryTime;
+                    DateTimeOffset absoluteExpiryTime = source.Properties.AbsoluteExpiryTime >= DateTimeOffset.MaxValue.UtcDateTime
+                        ? DateTimeOffset.MaxValue
+                        : source.Properties.AbsoluteExpiryTime.Value;
+
+                    message.Properties.AbsoluteExpiryTime = absoluteExpiryTime;
+
+                    // AbsoluteExpiryTime overwrites ttl coming in the header.
+                    // TTL in the header is a uint that stores the number of milliseconds.
+                    // Therefore the max TTL it can support is about 49 days (Uint32.MaxValue milliseconds)
```

Comment:
Do we need to guard against overflow/underflow here?

Reply:
I may be missing your point, but this comment (taken verbatim from Track 0 library) is saying that the max TTL value that will ever be stored in the header is uint32.MaxValue. The overflow/underflow is probably already handled in the AMQP library. In any case, we end up replacing the TTL based on the absolute expiry time. If for some reason the absolute expiry time is not set (which is not possible in Service Bus at least), only then would we actually use the TTL from the header. But at that point we have no idea what the original value was if the value is uint32.MaxValue.

Reply:
I will make the comment more clear.

Reply:
Ah okay I see your point now. We are not handling overflows when setting the uint32 field but we are always setting the creationTime and absoluteExpiryTime if the Ttl is set. If the Ttl is not set by the customer, but the default entity TTL is larger than uint32.MaxValue, we are relying on the fact that the broker correctly sets the AbsoluteExpiryTime. Still it would probably be a slight improvement to at least max out the TTL in the header rather than letting it wrap around.